external/vulkancts/modules/vulkan/conditional_rendering/vktConditionalDrawTests.cpp

```diff
@@ -0,0 +1,458 @@
+		m_data.push_back(Draw::VertexElementData(tcu::Vec4(	minX,	minY,	0.5f,	1.0f), tcu::RGBA::blue().toVec(), 0));
+		m_data.push_back(Draw::VertexElementData(tcu::Vec4(	maxX,	maxY,	0.5f,	1.0f), tcu::RGBA::blue().toVec(), 0));
+
+		m_data.push_back(Draw::VertexElementData(tcu::Vec4(	minX,	minY,	0.5f,	1.0f), tcu::RGBA::blue().toVec(), 0));
+		m_data.push_back(Draw::VertexElementData(tcu::Vec4(	maxX,	maxY,	0.5f,	1.0f), tcu::RGBA::blue().toVec(), 0));
+		m_data.push_back(Draw::VertexElementData(tcu::Vec4(	maxX,	minY,	0.5f,	1.0f), tcu::RGBA::blue().toVec(), 0));
+	}
+
+	m_data.push_back(Draw::VertexElementData(tcu::Vec4(	-1.0f,	1.0f,	0.0f,	1.0f), tcu::RGBA::red().toVec(), 0));
+	m_data.push_back(Draw::VertexElementData(tcu::Vec4(	-1.0f,	-1.0f,	0.0f,	1.0f), tcu::RGBA::red().toVec(), 0));
+	m_data.push_back(Draw::VertexElementData(tcu::Vec4(	1.0f,	1.0f,	0.0f,	1.0f), tcu::RGBA::red().toVec(), 0));
+
+	m_data.push_back(Draw::VertexElementData(tcu::Vec4(	-1.0f,	-1.0f,	0.0f,	1.0f), tcu::RGBA::red().toVec(), 0));
+	m_data.push_back(Draw::VertexElementData(tcu::Vec4(	1.0f,	1.0f,	0.0f,	1.0f), tcu::RGBA::red().toVec(), 0));
+	m_data.push_back(Draw::VertexElementData(tcu::Vec4(	1.0f,	-1.0f,	0.0f,	1.0f), tcu::RGBA::red().toVec(), 0));
+
+	for (deUint32 index = 0; index < m_data.size(); index++)
+	{
+		m_indexes.push_back(index);
+	}
+
+	initialize();
+
+	m_secondaryCmdBuffer = vk::allocateCommandBuffer(m_vk, m_context.getDevice(), *m_cmdPool, vk::VK_COMMAND_BUFFER_LEVEL_SECONDARY);
+};
+
+void ConditionalDraw::recordDraw(vk::VkCommandBuffer cmdBuffer)
+{
+	for (deUint32 drawIdx = 0; drawIdx < m_drawCalls; drawIdx++)
+	{
+		/* Indirect buffer has next layout:
+		 * goodCommand badCommand badCommand goodCommand badCommand badCommand ...
+		 */
+		const vk::VkDeviceSize indirectOffset = sizeof(vk::VkDrawIndirectCommand) * drawIdx * 3;
+		switch (m_command)
+		{
+			case DRAW_COMMAND_TYPE_DRAW:
+			{
+				m_vk.cmdDraw(cmdBuffer, 6, 1, 6 * drawIdx, 0);
+				break;
+			}
+			case DRAW_COMMAND_TYPE_DRAW_INDEXED:
+			{
+				m_vk.cmdDrawIndexed(cmdBuffer, 6, 1, 6 * drawIdx, 0, 0);
+				break;
+			}
+			case DRAW_COMMAND_TYPE_DRAW_INDIRECT:
+			{
+				m_vk.cmdDrawIndirect(cmdBuffer, m_indirectBuffer->object(), indirectOffset, 1, 0);
+				break;
+			}
+			case DRAW_COMMAND_TYPE_DRAW_INDEXED_INDIRECT:
+			{
+				m_vk.cmdDrawIndexedIndirect(cmdBuffer, m_indirectBuffer->object(), indirectOffset, 1, 0);
+				break;
+			}
+			case DRAW_COMMAND_TYPE_DRAW_INDIRECT_COUNT:
+			{
+				m_vk.cmdDrawIndirectCountKHR(	cmdBuffer,
+												m_indirectBuffer->object(), indirectOffset,
+												m_indirectCountBuffer->object(), 0, 3,
+												sizeof(vk::VkDrawIndirectCommand));
+				break;
+			}
+			case DRAW_COMMAND_TYPE_DRAW_INDEXED_INDIRECT_COUNT:
+			{
+				m_vk.cmdDrawIndexedIndirectCountKHR(cmdBuffer,
+													m_indirectBuffer->object(), indirectOffset,
+													m_indirectCountBuffer->object(), 0, 3,
+													sizeof(vk::VkDrawIndirectCommand));
```

Comment:
Should be VkDrawIndexedIndirectCommand instead of VkDrawIndirectCommand

Reply:
Yes, fixed related code.

Reply:
Also moved buffers initializations to separate functions, there were too many of them.